bin/level/app.rs

```diff
@@ -165,6 +213,36 @@ impl<R: gfx::Resources> Application<R> for LevelView<R> {
                 let rot = cgmath::Quaternion::from_angle_x(cgmath::Rad(1.0 * delta * dir));
                 self.cam.rot = self.cam.rot * rot;
             }
+            Input::DepQuant {dir} => {
+                let vec = cgmath::Vector3::unit_z();
+                self.cam.loc += 1000.0 * delta * dir * vec.normalize();
+                self.input = Input::Empty;
+            }
+            Input::PlaneQuant {dir} => {
+                let cgmath::Vector2{x, y} = dir;
+
+                let vec_x = self.cam.rot * cgmath::Vector3::unit_x() * -x;
```

Comment:
now I see what you want here, could look nicer with `cgmath::vec3(-x, 0.0, 0.0)`